tool/src/webapp/js/forum-summary.js

```diff
@@ -0,0 +1,70 @@
+var itemsToShow;
+$(function() {
+	//escape markup
+	String.prototype.escapeHTML = function(){
+		return (this.replace(/&/g, '&amp;').replace(/>/g, '&gt;').replace(/</g, '&lt;').replace(/"/g, '&quot;'));
+	};
+	itemsToShow = $('.forum-summary-div').find($('#numberOfConversations')).val();
+	//set the 'numberofconversations' in the hidden field for edit screen
+	$('#numberOfMessages').val(itemsToShow);
+	showForums();
+});
+
+function showForums(){
+	var forumsUrl = $('.forum-summary-site-url').text().replace(/'/g,"");
+	var messagesArray = [];
+	var errorText;
+	//Get forums
+	$.ajaxSetup({ cache: false,async: false });	//Disable caching for ajax requests and set to synchronous (because of nesting)
+	$.getJSON(forumsUrl+'.json', function(data_forum_collection) {
+		$(data_forum_collection["forums_collection"]).each(function(fc_index, fc_value){
+			$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'.json', function(data_forum) {
+				$(data_forum["topics"]).each(function(f_index,f_value){
+					$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'/topic/'+f_value['id']+'.json', function(data_topic) {
+						$(data_topic["threads"]).each(function(t_index,t_value){
+							$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'/topic/'+f_value['id']+'/message/'+t_value['messageId']+'.json', function(data_message) {
+								var author;
+								if(data_message['modifiedBy'].indexOf('(') > -1){
+									author = data_message['modifiedBy'].split('(')[0];
+								}else{
+									author = data_message['modifiedBy'];
+								}
+								//added forumId , topicId and messageId into the messageArray to link the message into the forum tool
+								var messageArray = [fc_value["entityId"],data_message['entityTitle'].escapeHTML(),data_message['entityURL'],data_message['lastModified'], author,f_value['id'],t_value['messageId']];
+								messagesArray.push(messageArray);
+								});
+							});
+						});
+					});
+				});
+			});
+			outputForums(messagesArray);
+			setTimeout('showForums()', 300000);
```

Comment:
As Jill has asked to refresh the list every 5 mins so the above is to reload the data every 300 secs or 5 mins